### PR TITLE
Fix onDelete constraints to align with production

### PR DESCRIPTION
**Pull request by spolu**
## Description

Fixes all onDelete to align with production. Some CASCADE remain here and there but all these are confirmed to be RESTRICT in production

Migration should be run in dev mode but won't be run in production.

## Risk

N/A (no-op)

## Deploy Plan

- N/A: no migration required as production already has these constraints

### CODE REVIEW

File: front/migrations/db/migration_138.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
+-- Migration created on Dec 31, 2024
+ALTER TABLE "user_metadata" ALTER COLUMN "userId" SET NOT NULL;ALTER TABLE "user_metadata"  ADD FOREIGN KEY ("userId") REFERENCES "users" ("id") ON DELETE RESTRICT ON UPDATE CASCADE;
```
Comment (flvndvd) [inline front/migrations/db/migration_138.sql]: Did you test this? IIRC this won't work as we need to delete the exiting ones.

Comment (spolu) [reply front/migrations/db/migration_138.sql]: It runs

Comment (spolu) [reply front/migrations/db/migration_138.sql]: It does create new constraints

Comment (spolu) [reply front/migrations/db/migration_138.sql]: Will just remove it

**Pull request merged**